homeassistant/components/mqtt/climate.py

```diff
@@ -226,6 +255,17 @@ def valid_preset_mode_configuration(config: ConfigType) -> ConfigType:
         ): cv.ensure_list,
         vol.Optional(CONF_FAN_MODE_STATE_TEMPLATE): cv.template,
         vol.Optional(CONF_FAN_MODE_STATE_TOPIC): valid_subscribe_topic,
+        vol.Optional(CONF_HUMIDITY_COMMAND_TEMPLATE): cv.template,
+        vol.Optional(CONF_HUMIDITY_COMMAND_TOPIC): valid_publish_topic,
+        vol.Optional(CONF_HUMIDITY_INITIAL, default=None): cv.positive_int,
+        vol.Optional(CONF_HUMIDITY_MIN, default=DEFAULT_MIN_HUMIDITY): vol.Coerce(
+            float
+        ),
+        vol.Optional(CONF_HUMIDITY_MAX, default=DEFAULT_MAX_HUMIDITY): vol.Coerce(
+            float
+        ),
+        vol.Optional(CONF_HUMIDITY_STATE_TEMPLATE): cv.template,
+        vol.Optional(CONF_HUMIDITY_STATE_TOPIC): valid_subscribe_topic,
```

Comment:
I suggest we add a check that `CONF_HUMIDITY_STATE_TOPIC` can only be configured if also `CONF_HUMIDITY_COMMAND_TOPIC` is configured.